extensions/ql-vscode/src/run-queries.ts

```diff
@@ -448,20 +469,30 @@ export async function compileAndRunQueryAgainstDatabase(
         " and choose CodeQL Query Server from the dropdown.");
     }
 
-    return {
-      query,
-      result: {
-        evaluationTime: 0,
-        resultType: messages.QueryResultType.OTHER_ERROR,
-        queryId: -1,
-        runId: -1,
-        message: "Query had compilation errors"
-      },
-      database: {
-        name: db.name,
-        databaseUri: db.databaseUri.toString(true)
-      },
-      options: historyItemOptions,
-    };
+    return createSyntheticResult(query, db, historyItemOptions, 'Query had compilation errors');
   }
 }
+
+function createSyntheticResult(
+  query: QueryInfo,
+  db: DatabaseItem,
+  historyItemOptions: QueryHistoryItemOptions,
+  message: string
+) {
+
+  return {
+    query,
+    result: {
+      evaluationTime: 0,
+      resultType: messages.QueryResultType.CANCELLATION,
```

Comment:
I think you want `resultType` to be a parameter of `createSyntheticResult`? This should still be `messages.QueryResultType.OTHER_ERROR` if we arrive here from the non-cancellation fallthrough (your diff's line 472).